test/functional/plugin/lsp/inlay_hint_spec.lua

```diff
@@ -258,3 +258,74 @@ describe('vim.lsp.inlay_hint', function()
     end)
   end)
 end)
+
+describe('Inlay hints are correctly rendered', function()
+  local text = dedent([[
+test text
+  ]])
+
+  local response = [==[
+  [
+      { "position": { "line": 0, "character": 0 }, "label": "0" },
+      { "position": { "line": 0, "character": 0 }, "label": "1" },
+      { "position": { "line": 0, "character": 0 }, "label": "2" },
+      { "position": { "line": 0, "character": 0 }, "label": "3" },
+      { "position": { "line": 0, "character": 0 }, "label": "4" }
+  ]
+  ]==]
+
+  local grid_without_inlay_hints = [[
+  test text                                         |
+   ^                                                 |
+                                                    |
+]]
+
+  local grid_with_inlay_hints = [[
+  {1:01234}test text                                    |
+   ^                                                 |
+                                                    |
+]]
+
+  --- @type test.functional.ui.screen
+  local screen
+  before_each(function()
+    clear_notrace()
+    screen = Screen.new(50, 3)
+    screen:attach()
+
+    exec_lua(create_server_definition)
+    exec_lua(
+      [[
+    local response = ...
+    server = _create_server({
+      capabilities = {
+        inlayHintProvider = true,
+      },
+      handlers = {
+        ['textDocument/inlayHint'] = function(_, _, callback)
+          callback(nil, vim.json.decode(response))
+        end,
+      }
+    })
+
+    bufnr = vim.api.nvim_get_current_buf()
+    vim.api.nvim_win_set_buf(0, bufnr)
+
+    client_id = vim.lsp.start({ name = 'dummy', cmd = server.cmd })
+  ]],
+      response
+    )
+    insert(text)
+  end)
+
+  it('when multiple inlay hints are at same character position', function()
```

Comment:
```suggestion
  it('renders hints with same position in received order, function()
```

Reply:
Done in [7248b73](https://github.com/neovim/neovim/pull/29707/commits/7248b7363ffdd00e0ac3288a92d4c0d2f7396a5c).